### PR TITLE
[ServiceBus] Update tracing

### PR DESCRIPTION
**Pull request by pvaneck**
Various tracing updates in this PR:  
 - A span will now be created upon calls to the service that settle messages.
 - Additional attributes added to spans:
     - `messaging.system` - messaging system (i.e., `servicebus`)
     - `messaging.operation` - type of operation (i.e., `publish`, `receive`, or `settle`)
     - `messaging.batch.message_count` - number of messages sent or received (if more than one)
  - All `send` spans now contain links to `message` spans. Now, `message` spans will no longer contain a link to the `send` span.

Maintained legacy attribute names for now.

[Reference overview](https://gist.github.com/lmolkova/e4215c0f44a49ef824983382762e6b92#other-api-calls-that-involve-communication-with-service) about tracing for AMQP

Part of: #25838

Ran some perf tests with tracing enabled (Opentelemetry SDK Tracer Provider) and disabled. Numbers below are from an average of five runs.

`SendQueueMessageBatchTest  --message-size 1024 --batch-size 100`
Tracing Off: 1102 ops/s
Tracing On: 1031 ops/s
Percent Change: **-6.4%**

`ReceiveQueueMessageBatchTest --message-size 2000 --num-messages 50 --preload 10000`
Tracing Off: 905.7 ops/s
Tracing On: 890.9 ops/s
Percent Change: **-1.6%**

### CODE REVIEW

**Review by swathipil — COMMENTED** [review]
lgtm, thanks @pvaneck! (left a couple comments + tested locally with an async sample, which works against these changes)

File: sdk/servicebus/azure-servicebus/CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,15 @@
...
 ### Other Changes
 
+ - Updated tracing ([#29995](https://github.com/Azure/azure-sdk-for-python/pull/29995)):
+   - A span will now be created upon calls to the service that settle messages.
+   - Additional attributes added to spans:
+     - `messaging.system` - messaging system (i.e., `servicebus`)
+     - `messaging.operation` - type of operation (i.e., `publish`, `receive`, or `settle`)
+     - `messaging.batch.message_count` - number of messages sent or received (if more than one)
+     - `messaging.source.name` added to receive spans in place of `message_bus.destination`
+   - All `send` spans now contain links to `message` spans. Now, `message` spans will no longer contain a link to the `send` span.
```
Comment (kashifkhan) [inline sdk/servicebus/azure-servicebus/CHANGELOG.md:20]: Are there any changes to any names that users should be aware of ?

File: sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_receiver.py [hunk 2]
```diff
@@ -468,17 +470,18 @@ def _settle_message_with_retry(
...
-            self._settle_message,
-            timeout=None,
-            message=message,
-            settle_operation=settle_operation,
-            dead_letter_reason=dead_letter_reason,
-            dead_letter_error_description=dead_letter_error_description,
-        )
-
-        message._settled = True
+        link = get_span_link_from_message(message)
```
[the remark this answers is not in the document]
Comment (pvaneck) [reply sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_receiver.py:473]: Passing in both `message._message` or just `message` seem to work in the case of received messages. Would `message._message` be preferred for reading message app properties?

File: sdk/servicebus/azure-servicebus/CHANGELOG.md (hunk 1 above)
Comment (pvaneck) [reply sdk/servicebus/azure-servicebus/CHANGELOG.md:20]: Span names stay the same. And pre-existing attribute names remain the same (at least for now). Several attributes are just added in addition. The otel plugin is intended to remap attributes to a specific set based on a specific otel schema version.

File: sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_receiver.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (swathipil) [reply sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_receiver.py:473]: you're definitely right, ignore me 🙂

**Pull request merged**